sdk/servicebus/Azure.Messaging.ServiceBus/src/Management/SharedAccessAuthorizationRule.cs

```diff
@@ -101,12 +102,14 @@ public string PrimaryKey
             {
                 if (string.IsNullOrWhiteSpace(value))
                 {
-                    throw new ArgumentNullException(nameof(PrimaryKey));
+                    throw new ArgumentNullException(null, nameof(PrimaryKey));
```

Comment:
Huh? Why do we need to do this?

Reply:
This was due to the FxCop error about not passing an argument. We discussed overriding the warning here.

Reply:
Looks like a client-side check so you can't set a `null` primary key. I'd assume that sending a null key won't work, and the process would fail during serialization.